packages/aws_common/lib/src/io/aws_file_platform_html.dart

```diff
@@ -81,10 +82,37 @@ class AWSFilePlatform extends AWSFile {
   final Stream<List<int>>? _stream;
   int? _size;
   Blob? _resolvedBlobFromPath;
-  String? _contentType;
+  String? _resolvedContentType;
+  bool _hasResolvedContentType = false;
 
   @override
-  String? get contentType => _contentType ?? super.contentType;
+  Future<String?> get contentType async {
```

Comment:
In Web file type can be determined via `Blob.type` hence not using `mime` package here.

Reply:
Does `Blob.type` just check the path or also the header bytes? Seems like there are cases below where it can't determine it.